lib/uksignal/signal.c

```diff
@@ -141,30 +147,34 @@ sigaction(int signum, const struct sigaction *act, struct sigaction *oldact)
 	return 0;
 }
 
-static sighandler_t uk_signal(int signum, sighandler_t handler, int sa_flags)
+#if UK_LIBC_SYSCALLS
+int sigaction(int signum, const struct sigaction *act, struct sigaction *oldact)
 {
+	return rt_sigaction(signum, act, oldact, (_NSIG / 8));
+}
+
+sighandler_t signal(int signum, sighandler_t handler)
+{
+	/* SA_RESTART <- BSD signal semantics */
 	struct sigaction old;
 	struct sigaction act = {
 		.sa_handler = handler,
-		.sa_flags = sa_flags
+		.sa_flags = SA_RESTART
 	};
 
-	if (sigaction(signum, &act, &old) < 0)
+	if (rt_sigaction(signum, &act, &old, (_NSIG / 8)) < 0)
 		return SIG_ERR;
 
 	if (old.sa_flags & SA_SIGINFO)
 		return NULL;
 	else
 		return old.sa_handler;
 }
+#endif
```

Comment:
Please always provide the macro you tested as comment with the `endif`
```suggestion
#endif /* UK_LIBC_SYSCALLS */
```